server/src/useCases/report/productTestCases/handleGetProductDetailsRequest.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable object-curly-newline */
+/* eslint-disable no-console */
+
+module.exports = class ReportGetProductDetailsRequestHandler {
+  constructor(request, response) {
+    this.req = request;
+    this.res = response;
+    this.dbConnect = request.app.locals.reportCollection;
+  }
+
+  async getProductDetails(repository) {
+    const { limit, offset } = this.req.query;
+    const { id } = this.req.params;
+
+    repository.aggregateByProductId({ id }, async (err, result) => {
+      if (err) {
+        console.error(err);
+        this.res
+          .status(500)
+          .send(`Failed to fetch detailed report summary. Details: \n\t${err}\nPlease contact administrator.`);
+        return;
+      }
+
+      const aggregatedResult = result[0];
+
+      let tempItem = {};
+      aggregatedResult.data = aggregatedResult.data.reduce((items, item) => {
+        const { uri, passed, method, endpoint } = item;
+
+        if (tempItem.uri !== uri) {
+          // sliced by @method= length
+          item.method = method.length > 0 ? method[0].slice(8) : '';
+          // sliced by @endpoint= length
+          item.endpoint = endpoint.length > 0 ? endpoint[0].slice(10) : '';
+
+          items.push(item);
+        } else if (tempItem.passed !== passed && !passed) {
+          items.find((i) => i.uri === uri).passed = passed;
+        }
+
+        tempItem = item;
+        return items;
+      }, []);
+
+      Object.assign(aggregatedResult, { count: aggregatedResult.data.length });
+
+      aggregatedResult.data = offset !== undefined
```

Comment:
wouldn't it be more effective to reduce before slicing?

Moreover, shouldn't we be paginating in the database instead of in-memory? What data volume do we expect here?

Reply:
In general this whole endpoint seems very weird. Can you explain what are you trying to return? As in, how the json should look like and what's the point of this?

I've take a look at the ticket, but there's nothing there about these aggregations, slicing names, etc.

Reply:
![image](https://user-images.githubusercontent.com/119664031/226673595-dd483fc0-c9cc-4a72-8a3f-f9d7c9351ad5.png)

It should return all necessary data for this page.

Reply:
I sliced it because in the database ```endpoint``` and ```method``` are stored as tags from gherkin file scenario.
etc ```@method=GET```

Reply:
regarding the data:

- is method used anywhere for example? if not we should probably drop it

If you are building a new data model here and not returning as is from the database - you should explicitly build a new model in the reduce method, not reuse what you have (but in a way that switches up some fields)

Reply:
one more thing - if you build a util that will do this string clean up for you, you can just use it in the reduce method as you're putting this data together - in order to save the iterations needed

Reply:
method is used in 2 places - if you mean ```@method=``` 

I added slicing those strings to mondodb aggregation. 

I'm not sure if i completely understood ```If you are building a new data model here and not returning as is from the database - you should explicitly build a new model in the reduce method, not reuse what you have (but in a way that switches up some fields)``` could you provide more information? For now i decided to build the object after pagination.